### ISSUE TITLE
cElementTree behaves differently compared to ElementTree

### ISSUE THREAD

**Issue by c0de0930-f94f-419e-a832-378a5ebc6264** [python/cpython#75648]
BPO | [31467](https://bugs.python.org/issue31467)
--- | :---
Nosy | @scoder, @serhiy-storchaka

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = <Date 2017-09-14.12:24:41.059>
created_at = <Date 2017-09-14.10:06:21.457>
labels = ['expert-XML', 'type-bug', 'library']
title = 'cElementTree behaves differently compared to ElementTree'
updated_at = <Date 2018-05-28.16:08:46.949>
user = 'https://bugs.python.org/poojariravi'
```

bugs.python.org fields:
```python
activity = <Date 2018-05-28.16:08:46.949>
actor = 'serhiy.storchaka'
assignee = 'none'
closed = True
closed_date = <Date 2017-09-14.12:24:41.059>
closer = 'poojariravi'
components = ['Library (Lib)', 'XML']
creation = <Date 2017-09-14.10:06:21.457>
creator = 'poojariravi'
dependencies = []
files = []
hgrepos = []
issue_num = 31467
keywords = []
message_count = 5.0
messages = ['302164', '302165', '302166', '302167', '302168']
nosy_count = 4.0
nosy_names = ['scoder', 'eli.bendersky', 'serhiy.storchaka', 'poojariravi']
pr_nums = []
priority = 'normal'
resolution = 'wont fix'
stage = 'resolved'
status = 'closed'
superseder = None
type = 'behavior'
url = 'https://bugs.python.org/issue31467'
versions = ['Python 2.7']
```

</p></details>

**Comment by c0de0930-f94f-419e-a832-378a5ebc6264** [comment]
I have started learning python and I see below snippet fails. I do not understand how to fix the issue and assume this might be a bug in multiprocessing library.

I use Python 2.7.12


My python snippet

```py
import xml.etree.ElementTree as ET    # WORKS
import xml.etree.cElementTree as ET    # DOES NOT WORK
import multiprocessing

tree = ET.parse('country_data.xml')
root = tree.getroot()

manager = multiprocessing.Manager()
elems_saved = manager.dict()
```


elems_saved["1"]=root


/tmp/Python$ cat country_data.xml 
\<?xml version="1.0"?\>
\<data\>
    \<country name="Liechtenstein"\>
        \<rank\>1\</rank\>
        \<year\>2008\</year\>
        \<gdppc\>141100\</gdppc\>
        \<neighbor name="Austria" direction="E"/\>
        \<neighbor name="Switzerland" direction="W"/\>
    \</country\>
\</data\>
/tmp/Python$


OUTPUT

```pytb
/tmp/Python$ python testxml.py 
Traceback (most recent call last):
  File "testxml.py", line 10, in <module>
    elems_saved["1"]=root
  File "<string>", line 2, in __setitem__
  File "/usr/lib/python2.7/multiprocessing/managers.py", line 758, in _callmethod
    conn.send((self._id, methodname, args, kwds))
TypeError: expected string or Unicode object, NoneType found
```


Please let me know if I am wrong

**Comment by serhiy-storchaka** [comment]
This error is from cPickle.

```pycon
>>> import xml.etree.cElementTree as ET
>>> root = ET.XML('<data/>')
>>> import pickle
>>> pickle.dumps(root)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib/python2.7/pickle.py", line 1380, in dumps
    Pickler(file, protocol).dump(obj)
  File "/usr/lib/python2.7/pickle.py", line 224, in dump
    self.save(obj)
  File "/usr/lib/python2.7/pickle.py", line 331, in save
    self.save_reduce(obj=obj, *rv)
  File "/usr/lib/python2.7/pickle.py", line 400, in save_reduce
    save(func)
  File "/usr/lib/python2.7/pickle.py", line 286, in save
    f(self, obj) # Call unbound method with explicit self
  File "/usr/lib/python2.7/pickle.py", line 754, in save_global
    (obj, module, name))
pickle.PicklingError: Can't pickle <function copyelement at 0x7f8acaf5b758>: it's not found as __main__.copyelement
>>> import cPickle
>>> cPickle.dumps(root)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: expected string or Unicode object, NoneType found
```


cElementTree.Element is not pickleable in Python 2. And I think it is too later to add this feature in 2.7. The obvious solution -- use Python 3.

**Comment by c0de0930-f94f-419e-a832-378a5ebc6264** [comment]
Thanks for the quick update. It helped us to understand the issue.

We will check of how to proceed further in our implementation.

/ Ravi

**Comment by c0de0930-f94f-419e-a832-378a5ebc6264** [comment]
Just now I have figured out that , in the documentation it was mentioned that cElementTree module is deprecated in Python 3.3 version. 

"The xml.etree.ElementTree module implements a simple and efficient API for parsing and creating XML data.

Changed in version 3.3: This module will use a fast implementation whenever available. The xml.etree.cElementTree module is deprecated."

https://docs.python.org/3/library/xml.etree.elementtree.html


In that case, will your resolution comment still holds valid ??
If so then, how to use cElementTree in my usecase
